public/app/features/dashboard-scene/settings/DeleteProvisionedDashboardForm.tsx

```diff
@@ -0,0 +1,147 @@
+import { useForm, FormProvider } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom-v5-compat';
+
+import { AppEvents } from '@grafana/data';
+import { Trans, t } from '@grafana/i18n';
+import { getAppEvents } from '@grafana/runtime';
+import { Alert, Button, Drawer, Stack } from '@grafana/ui';
+import { useDeleteRepositoryFilesWithPathMutation } from 'app/api/clients/provisioning/v0alpha1';
+import { PROVISIONING_URL } from 'app/features/provisioning/constants';
+
+import { DashboardEditFormSharedFields } from '../components/Provisioned/DashboardEditFormSharedFields';
+import { ProvisionedDashboardFormData } from '../saving/shared';
+import { DashboardScene } from '../scene/DashboardScene';
+import { useProvisionedRequestHandler } from '../utils/useProvisionedRequestHandler';
+
+export interface Props {
+  dashboard: DashboardScene;
+  defaultValues: ProvisionedDashboardFormData;
+  readOnly: boolean;
+  isGitHub: boolean;
+  isNew?: boolean;
+  workflowOptions: Array<{ label: string; value: string }>;
+  loadedFromRef?: string;
+  onDismiss: () => void;
+}
+
+/**
+ * @description
+ * Drawer component for deleting a git provisioned dashboard.
+ */
+export function DeleteProvisionedDashboardForm({
+  dashboard,
+  defaultValues,
+  loadedFromRef,
+  readOnly,
+  isGitHub,
+  isNew,
+  workflowOptions,
+  onDismiss,
+}: Props) {
+  const methods = useForm<ProvisionedDashboardFormData>({ defaultValues });
+  const { editPanel: panelEditor } = dashboard.useState();
+  const { handleSubmit, watch } = methods;
+
+  const [ref, workflow] = watch(['ref', 'workflow']);
+  const [deleteRepoFile, request] = useDeleteRepositoryFilesWithPathMutation();
+
+  const handleSubmitForm = async ({ repo, path, comment }: ProvisionedDashboardFormData) => {
+    if (!repo || !path) {
+      console.error('Missing required fields for deletion:', { repo, path });
+      return;
+    }
+
+    // If writing to the original branch, use the loaded reference; otherwise, use the selected ref.
+    const branchRef = workflow === 'write' ? loadedFromRef : ref;
+    const commitMessage = comment || `Delete dashboard: ${dashboard.state.title}`;
+
+    deleteRepoFile({
+      name: repo,
+      path: path,
+      ref: branchRef,
+      message: commitMessage,
+    });
+  };
+
+  const navigate = useNavigate();
+
+  const onRequestError = (error: unknown) => {
+    getAppEvents().publish({
+      type: AppEvents.alertError.name,
+      payload: [t('dashboard-scene.delete-provisioned-dashboard-form.api-error', 'Failed to delete dashboard'), error],
+    });
+  };
+
+  const onWriteSuccess = () => {
+    panelEditor?.onDiscard();
+    onDismiss();
+    navigate('/dashboards');
```

Comment:
```suggestion
    // TODO reset search state instead
    window.location.href = '/dashboards';
```
We need to use the full page refresh here, otherwise the deleted dashboard is still shown in the search result after deleting. Related to https://github.com/grafana/git-ui-sync-project/issues/102.

Reply:
good to know! thanks : )